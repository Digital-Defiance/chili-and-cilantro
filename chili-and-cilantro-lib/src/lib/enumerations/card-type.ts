export enum CardType {
  CHILI = 'chili',
  CILANTRO = 'cilantro',
}

export default CardType;