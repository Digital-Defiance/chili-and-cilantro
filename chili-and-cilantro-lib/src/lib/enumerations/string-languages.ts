export enum StringLanguages {
  EnglishUS = 'English (US)',
  EnglishUK = 'English (UK)',
  French = 'Français',
  MandarinChinese = '中文',
  Spanish = 'Español',
  Ukrainian = 'Український',
}
