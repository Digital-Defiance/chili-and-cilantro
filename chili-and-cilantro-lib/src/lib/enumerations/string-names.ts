/**
 * String names for localization
 * An underscore converts to a period which causes nested object access
 * WARNING: You cannot have a key with name A and then have A_anything because it will not be able to create a nested object on a string
 * For consistency, the left side should match the right, give or take CamelCase on the left and snake case (e.g. common_blahBlah) on the right
 */
export enum StringNames {
  AccountError_Message = 'accountError_message',
  AccountError_Title = 'accountError_title',
  ChangePassword_ChangePasswordButton = 'changePassword_changePasswordButton',
  Common_ChangePassword = 'common_changePassword',
  Common_CurrentPassword = 'common_currentPassword',
  Common_Email = 'common_email',
  Common_GoToSplash = 'common_goToSplash',
  Common_Logo = 'common_logo',
  Common_NewPassword = 'common_newPassword',
  Common_ConfirmNewPassword = 'common_confirmNewPassword',
  Common_Dashboard = 'common_dashboard',
  Common_Loading = 'common_loading',
  Common_Password = 'common_password',
  Common_ReturnToKitchen = 'common_returnToKitchen',
  Common_Site = 'common_site',
  Common_StartCooking = 'common_startCooking',
  Common_Tagline = 'common_tagline',
  Common_Unauthorized = 'common_unauthorized',
  Common_UnexpectedError = 'common_unexpectedError',
  Common_Username = 'common_username',
  Error_AccountStatusIsDeleted = 'error_accountStatusIsDeleted',
  Error_AccountStatusIsLocked = 'error_accountStatusIsLocked',
  Error_AccountStatusIsPendingEmailVerification = 'error_accountStatusIsPendingEmailVerification',
  Dashboard_GamesCreated = 'dashboard_gamesCreated',
  Dashboard_GamesParticipating = 'dashboard_gamesParticipating',
  Dashboard_NoGames = 'dashboard_noGames',
  Dashboard_Title = 'dashboard_title',
  ForgotPassword_ForgotPassword = 'forgotPassword_forgotPassword',
  ForgotPassword_InvalidToken = 'forgotPassword_invalidToken',
  ForgotPassword_ResetPassword = 'forgotPassword_resetPassword',
  ForgotPassword_SendResetToken = 'forgotPassword_sendResetToken',
  ForgotPassword_Success = 'forgotPassword_success',
  ForgotPassword_Title = 'forgotPassword_title',
  Game_CreateGame = 'game_createGame',
  Game_CreateGameSuccess = 'game_createGameSuccess',
  Game_JoinGame = 'game_joinGame',
  Game_JoinGameSuccess = 'game_joinGameSuccess',
  Game_ListSuccess = 'game_listSuccess',
  KeyFeatures_Title = 'keyFeatures_title',
  KeyFeatures_1 = 'keyFeatures_1',
  KeyFeatures_2 = 'keyFeatures_2',
  KeyFeatures_3 = 'keyFeatures_3',
  KeyFeatures_4 = 'keyFeatures_4',
  KeyFeatures_5 = 'keyFeatures_5',
  KeyFeatures_6 = 'keyFeatures_6',
  KeyFeatures_7 = 'keyFeatures_7',
  KeyFeatures_8 = 'keyFeatures_8',
  LanguageUpdate_Success = 'languageUpdate_success',
  Login_LoginButton = 'login_loginButton',
  Login_Progress = 'login_progress',
  Login_ResentPasswordFailure = 'login_resentPasswordFailure',
  Login_ResentPasswordSuccess = 'login_resentPasswordSuccess',
  Login_Title = 'login_title',
  Login_UsernameOrEmailRequired = 'login_usernameOrEmailRequired',
  LogoutButton = 'logoutButton',
  RegisterButton = 'registerButton',
  Splash_Description = 'splash_description',
  Splash_HowToPlay = 'splash_howToPlay',
  ValidationError = 'validationError',
  Validation_InvalidEmail = 'validation_invalidEmail',
  Validation_InvalidLanguage = 'validation_invalidLanguage',
  Validation_InvalidTimezone = 'validation_invalidTimezone',
  Validation_InvalidToken = 'validation_invalidToken',
  Validation_PasswordRegexErrorTemplate = 'validation_passwordRegexErrorTemplate',
  Validation_CurrentPasswordRequired = 'validation_currentPasswordRequired',
  Validation_PasswordsDifferent = 'validation_passwordsDifferent',
  Validation_NewPasswordRequired = 'validation_newPasswordRequired',
  Validation_PasswordMatch = 'validation_passwordMatch',
  Validation_ConfirmNewPassword = 'validation_confirmNewPassword',
  Validation_Required = 'validation_required',
  Validation_UsernameRegexErrorTemplate = 'validation_usernameRegexErrorTemplate',
}
