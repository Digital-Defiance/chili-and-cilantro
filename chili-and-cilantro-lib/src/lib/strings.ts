import { StringLanguages } from './enumerations/string-languages';
import { MasterStringsCollection } from './shared-types';
import BritishEnglishStrings from './strings/english-uk';
import AmericanEnglishStrings from './strings/english-us';
import FrenchStrings from './strings/french';
import MandarinStrings from './strings/mandarin';
import SpanishStrings from './strings/spanish';
import UkrainianStrings from './strings/ukrainian';

export const Strings: MasterStringsCollection = {
  [StringLanguages.EnglishUS]: AmericanEnglishStrings,
  [StringLanguages.EnglishUK]: BritishEnglishStrings,
  [StringLanguages.French]: FrenchStrings,
  [StringLanguages.MandarinChinese]: MandarinStrings,
  [StringLanguages.Spanish]: SpanishStrings,
  [StringLanguages.Ukrainian]: UkrainianStrings,
};
