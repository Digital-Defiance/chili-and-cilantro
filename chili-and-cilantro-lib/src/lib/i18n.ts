import { AccountStatusTypeTranslations } from './enumeration-translations/account-status-type';
import { ActionTypeTranslations } from './enumeration-translations/action-type';
import { ActionTypePastTenseTranslations } from './enumeration-translations/action-type-past-tense';
import { CardTypeTranslations } from './enumeration-translations/card-type';
import { ChallengeResponseTranslations } from './enumeration-translations/challenge-response';
import { ChallengeResponsePastTenseTranslations } from './enumeration-translations/challenge-response-past-tense';
import { ChefStateTranslations } from './enumeration-translations/chef-state';
import { EmailTokenTypeTranslations } from './enumeration-translations/email-token-type';
import { GamePhaseTranslations } from './enumeration-translations/game-phase';
import { QuitGameReasonTranslations } from './enumeration-translations/quit-game-reason';
import { TurnActionTranslations } from './enumeration-translations/turn-action';
import { TurnActionPastTenseTranslations } from './enumeration-translations/turn-action-past-tense';
import { AccountStatusTypeEnum } from './enumerations/account-status-type';
import ActionType from './enumerations/action-type';
import CardType from './enumerations/card-type';
import ChallengeResponse from './enumerations/challenge-response';
import ChefState from './enumerations/chef-state';
import { EmailTokenType } from './enumerations/email-token-type';
import GamePhase from './enumerations/game-phase';
import QuitGameReason from './enumerations/quit-game-reason';
import { StringLanguages } from './enumerations/string-languages';
import { StringNames } from './enumerations/string-names';
import { TranslatableEnumType } from './enumerations/translatable-enum';
import TurnAction from './enumerations/turn-action';
import { ILanguageContext } from './interfaces/language-context';
import { LanguageCodes } from './language-codes';
import { DefaultLanguage, StringsCollection } from './shared-types';
import { Strings } from './strings';

export const buildNestedI18n = (
  strings: StringsCollection,
): Record<string, any> => {
  const result: Record<string, any> = {};

  Object.entries(strings).forEach(([key, value]) => {
    const keys = key.split('_');
    let current = result;

    keys.forEach((k, index) => {
      if (index === keys.length - 1) {
        // Assign the value at the deepest level
        if (typeof current[k] === 'object' && current[k] !== null) {
          throw new Error(
            `Key conflict detected: Cannot assign string to key '${k}' because it's already used as an object.`,
          );
        }
        current[k] = value;
      } else {
        // Create nested objects if they don't exist
        if (!(k in current)) {
          current[k] = {};
        } else if (typeof current[k] !== 'object' || current[k] === null) {
          // Conflict detected
          throw new Error(
            `Key conflict detected: Key '${k}' is assigned both a value and an object.`,
          );
        }
        current = current[k];
      }
    });
  });

  return result;
};

export const buildNestedI18nForLanguage = (language: StringLanguages) => {
  if (!Strings[language]) {
    throw new Error(`Strings not found for language: ${language}`);
  }

  return buildNestedI18n(Strings[language]);
};

export const stringNameToI18nKey = (name: StringNames) =>
  name.replace('_', '.'); // only replace the first underscore

export const translate = (
  name: StringNames,
  language?: StringLanguages,
): string => {
  const lang = language ?? GlobalLanguageContext.language;
  if (!Strings[lang]) {
    console.warn(`Language ${lang} not found in Strings`);
    return name; // Fallback to the string name itself
  }
  if (!(name in Strings[lang])) {
    console.warn(`String ${name} not found for language ${lang}`);
    return name; // Fallback to the string name itself
  }
  return Strings[lang][name];
};

export type TranslatableEnum =
  | { type: TranslatableEnumType.AccountStatus; value: AccountStatusTypeEnum }
  | { type: TranslatableEnumType.ActionType; value: ActionType }
  | { type: TranslatableEnumType.ActionTypePastTense; value: ActionType }
  | { type: TranslatableEnumType.CardType; value: CardType }
  | { type: TranslatableEnumType.ChallengeResponse; value: ChallengeResponse }
  | {
      type: TranslatableEnumType.ChallengeResponsePastTense;
      value: ChallengeResponse;
    }
  | { type: TranslatableEnumType.ChefState; value: ChefState }
  | { type: TranslatableEnumType.EmailTokenType; value: EmailTokenType }
  | { type: TranslatableEnumType.GamePhase; value: GamePhase }
  | { type: TranslatableEnumType.QuitGameReason; value: QuitGameReason }
  | { type: TranslatableEnumType.TurnAction; value: TurnAction }
  | { type: TranslatableEnumType.TurnActionPastTense; value: TurnAction };

export type TranslationsMap = {
  [TranslatableEnumType.AccountStatus]: {
    [key in StringLanguages]: { [key in AccountStatusTypeEnum]: string };
  };
  [TranslatableEnumType.ActionType]: {
    [key in StringLanguages]: { [key in ActionType]: string };
  };
  [TranslatableEnumType.ActionTypePastTense]: {
    [key in StringLanguages]: { [key in ActionType]: string };
  };
  [TranslatableEnumType.CardType]: {
    [key in StringLanguages]: { [key in CardType]: string };
  };
  [TranslatableEnumType.ChallengeResponse]: {
    [key in StringLanguages]: {
      [key in ChallengeResponse]: string;
    };
  };
  [TranslatableEnumType.ChallengeResponsePastTense]: {
    [key in StringLanguages]: { [key in ChallengeResponse]: string };
  };
  [TranslatableEnumType.ChefState]: {
    [key in StringLanguages]: { [key in ChefState]: string };
  };
  [TranslatableEnumType.EmailTokenType]: {
    [key in StringLanguages]: { [key in EmailTokenType]: string };
  };
  [TranslatableEnumType.GamePhase]: {
    [key in StringLanguages]: { [key in GamePhase]: string };
  };
  [TranslatableEnumType.QuitGameReason]: {
    [key in StringLanguages]: { [key in QuitGameReason]: string };
  };
  [TranslatableEnumType.TurnAction]: {
    [key in StringLanguages]: { [key in TurnAction]: string };
  };
  [TranslatableEnumType.TurnActionPastTense]: {
    [key in StringLanguages]: { [key in TurnAction]: string };
  };
};

export const translationsMap: TranslationsMap = {
  [TranslatableEnumType.AccountStatus]: AccountStatusTypeTranslations,
  [TranslatableEnumType.ActionType]: ActionTypeTranslations,
  [TranslatableEnumType.ActionTypePastTense]: ActionTypePastTenseTranslations,
  [TranslatableEnumType.CardType]: CardTypeTranslations,
  [TranslatableEnumType.ChallengeResponse]: ChallengeResponseTranslations,
  [TranslatableEnumType.ChallengeResponsePastTense]:
    ChallengeResponsePastTenseTranslations,
  [TranslatableEnumType.ChefState]: ChefStateTranslations,
  [TranslatableEnumType.EmailTokenType]: EmailTokenTypeTranslations,
  [TranslatableEnumType.GamePhase]: GamePhaseTranslations,
  [TranslatableEnumType.QuitGameReason]: QuitGameReasonTranslations,
  [TranslatableEnumType.TurnAction]: TurnActionTranslations,
  [TranslatableEnumType.TurnActionPastTense]: TurnActionPastTenseTranslations,
};

export const translateEnum = (
  { type, value }: TranslatableEnum,
  language?: StringLanguages,
): string => {
  const lang = language ?? GlobalLanguageContext.language;
  const translations = translationsMap[type];
  if (translations && translations[lang]) {
    const enumTranslations = translations[lang] as Record<string, string>;
    if (enumTranslations[value]) {
      return enumTranslations[value];
    }
  }
  throw new Error(`Unknown enum value: ${value} for type: ${type}`);
};

export const GlobalLanguageContext: ILanguageContext = {
  language: DefaultLanguage,
};

export function getLanguageCode(language: string): StringLanguages {
  for (const [key, value] of Object.entries(LanguageCodes)) {
    if (value === language) {
      return key as StringLanguages;
    }
  }
  throw new Error(`Unknown language code: ${language}`);
}
