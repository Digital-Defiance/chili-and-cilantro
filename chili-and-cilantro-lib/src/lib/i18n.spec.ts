import { StringLanguages } from './enumerations/string-languages';
import { StringNames } from './enumerations/string-names';
import { buildNestedI18n } from './i18n'; // Adjust the import path as needed
import { StringsCollection } from './shared-types';
import { Strings } from './strings';

describe('buildNestedI18n', () => {
  it('should handle flat strings', () => {
    const input: StringsCollection = {
      [StringNames.Common_Site]: 'Chili and Cilantro',
      [StringNames.Common_Tagline]: 'A Spicy Bluffing Game',
    } as unknown as StringsCollection;
    const expected = {
      common: {
        site: 'Chili and Cilantro',
        tagline: 'A Spicy Bluffing Game',
      },
    };
    expect(buildNestedI18n(input)).toEqual(expected);
  });

  it('should create nested objects for keys with underscores', () => {
    const input = {
      dashboard_title: 'Dashboard',
      dashboard_yourDashboard: 'Your Dashboard',
    } as unknown as StringsCollection;
    const expected = {
      dashboard: {
        title: 'Dashboard',
        yourDashboard: 'Your Dashboard',
      },
    };
    expect(buildNestedI18n(input)).toEqual(expected);
  });

  it('should handle numeric keys', () => {
    const input = {
      keyFeatures_label: 'Key Features',
      keyFeatures_1: 'Feature 1',
      keyFeatures_2: 'Feature 2',
    } as unknown as StringsCollection;
    const expected = {
      keyFeatures: {
        label: 'Key Features',
        '1': 'Feature 1',
        '2': 'Feature 2',
      },
    };
    expect(buildNestedI18n(input)).toEqual(expected);
  });

  it('should handle mixed nested and flat keys', () => {
    const input = {
      site: 'CurseFund',
      validation_usernameMinLength: 'Username must be at least 3 characters',
      validation_passwordMinLength: 'Password must be at least 8 characters',
      errors_404: 'Page not found',
    } as unknown as StringsCollection;
    const expected = {
      site: 'CurseFund',
      validation: {
        usernameMinLength: 'Username must be at least 3 characters',
        passwordMinLength: 'Password must be at least 8 characters',
      },
      errors: {
        '404': 'Page not found',
      },
    };
    expect(buildNestedI18n(input)).toEqual(expected);
  });

  it('should handle empty input', () => {
    const input = {} as StringsCollection;
    const expected = {};
    expect(buildNestedI18n(input)).toEqual(expected);
  });

  it('should not modify original input', () => {
    const input = {
      a_b: 'Value',
      c: 'Another Value',
    } as unknown as StringsCollection;
    const originalInput = { ...input };
    buildNestedI18n(input);
    expect(input).toEqual(originalInput);
  });

  it('should throw an error when a key is both a string and an object', () => {
    const input = {
      keyFeatures: 'Key Features',
      keyFeatures_1: 'Feature 1',
      keyFeatures_2: 'Feature 2',
    } as unknown as StringsCollection;

    expect(() => buildNestedI18n(input)).toThrowError(
      "Key conflict detected: Key 'keyFeatures' is assigned both a value and an object.",
    );
  });

  it.each(Object.values(StringLanguages))(
    'should handle %s language',
    (language) => {
      const stringCollection = Strings[language];
      let nestedI18n;
      expect(() => {
        nestedI18n = buildNestedI18n(stringCollection);
      }).not.toThrow();
      expect(nestedI18n).toBeTruthy();
      expect(nestedI18n).toMatchSnapshot();
    },
  );
});
