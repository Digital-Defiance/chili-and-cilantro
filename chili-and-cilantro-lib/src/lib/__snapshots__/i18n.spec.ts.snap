// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildNestedI18n should handle English (UK) language 1`] = `
{
  "changePassword": {
    "changePasswordButton": "Change Password",
  },
  "common": {
    "changePassword": "Change Password",
    "confirmNewPassword": "Confirm New Password",
    "currentPassword": "Current Password",
    "dashboard": "Dashboard",
    "loading": "Loading...",
    "logo": "logo",
    "newPassword": "New Password",
    "returnToKitchen": "Return to Kitchen",
    "site": "Chili and Cilantro",
    "startCooking": "Start Cooking",
    "tagline": "A Spicy Bluffing Game",
    "unauthorized": "Unauthorized",
    "unexpectedError": "Unexpected Error",
  },
  "dashboard": {
    "gamesCreated": "Games You've Created",
    "gamesParticipating": "Games You're Participating in",
    "noGames": "No games available.",
    "title": "Your Dashboard",
  },
  "forgotPassword": {
    "title": "Forgot Password",
  },
  "game": {
    "createGame": "Create Game",
    "listSuccess": "Retrieved game list successfully",
  },
  "keyFeatures": {
    "1": "Exciting bluffing gameplay with a culinary twist",
    "2": "Strategic bidding and card placement",
    "3": "Quick to learn, challenging to master",
    "4": "Supports 2 or more players",
    "5": "Rounds of suspenseful card flipping",
    "6": "Risk management: avoid the dreaded chili!",
    "7": "First to season two dishes or last chef standing wins",
    "8": "Perfect for game nights and family gatherings",
    "title": "Key Features",
  },
  "languageUpdate": {
    "success": "Language updated successfully",
  },
  "login": {
    "loginButton": "Login",
  },
  "logoutButton": "Logout",
  "registerButton": "Register",
  "splash": {
    "description": "In Chili and Cilantro, aspiring chefs compete to create the perfect dish. Your goal is to add just the right amount of cilantro without ruining it with a scorching chili. Be the first to successfully season two dishes or be the last chef standing to win!",
    "howToPlay": "How to Play",
  },
  "validation": {
    "confirmNewPassword": "Confirm new password is required",
    "currentPasswordRequired": "Current password is required",
    "invalidEmail": "Invalid Email",
    "invalidLanguage": "Invalid Language",
    "invalidToken": "Invalid Token",
    "newPasswordRequired": "New password is required",
    "passwordMatch": "Password and confirm must match",
    "passwordRegexError": "Password must be at least 8 characters long and include at least one letter, one number, and one special character (!@#$%^&*()_+-=[]{};':"|,.<>/?)",
    "passwordsDifferent": "New password must be different than the current password",
  },
  "validationError": "Validation Error",
}
`;

exports[`buildNestedI18n should handle English (US) language 1`] = `
{
  "changePassword": {
    "changePasswordButton": "Change Password",
  },
  "common": {
    "changePassword": "Change Password",
    "confirmNewPassword": "Confirm New Password",
    "currentPassword": "Current Password",
    "dashboard": "Dashboard",
    "loading": "Loading...",
    "logo": "logo",
    "newPassword": "New Password",
    "returnToKitchen": "Return to Kitchen",
    "site": "Chili and Cilantro",
    "startCooking": "Start Cooking",
    "tagline": "A Spicy Bluffing Game",
    "unauthorized": "Unauthorized",
    "unexpectedError": "Unexpected Error",
  },
  "dashboard": {
    "gamesCreated": "Games You've Created",
    "gamesParticipating": "Games You're Participating in",
    "noGames": "No games available.",
    "title": "Your Dashboard",
  },
  "forgotPassword": {
    "title": "Forgot Password",
  },
  "game": {
    "createGame": "Create Game",
    "listSuccess": "Retrieved game list successfully",
  },
  "keyFeatures": {
    "1": "Exciting bluffing gameplay with a culinary twist",
    "2": "Strategic bidding and card placement",
    "3": "Quick to learn, challenging to master",
    "4": "Supports 2 or more players",
    "5": "Rounds of suspenseful card flipping",
    "6": "Risk management: avoid the dreaded chili!",
    "7": "First to season two dishes or last chef standing wins",
    "8": "Perfect for game nights and family gatherings",
    "title": "Key Features",
  },
  "languageUpdate": {
    "success": "Language updated successfully",
  },
  "login": {
    "loginButton": "Login",
  },
  "logoutButton": "Logout",
  "registerButton": "Register",
  "splash": {
    "description": "In Chili and Cilantro, aspiring chefs compete to create the perfect dish. Your goal is to add just the right amount of cilantro without ruining it with a scorching chili. Be the first to successfully season two dishes or be the last chef standing to win!",
    "howToPlay": "How to Play",
  },
  "validation": {
    "confirmNewPassword": "Confirm new password is required",
    "currentPasswordRequired": "Current password is required",
    "invalidEmail": "Invalid Email",
    "invalidLanguage": "Invalid Language",
    "invalidToken": "Invalid Token",
    "newPasswordRequired": "New password is required",
    "passwordMatch": "Password and confirm must match",
    "passwordRegexError": "Password must be at least 8 characters long and include at least one letter, one number, and one special character (!@#$%^&*()_+-=[]{};':"|,.<>/?)",
    "passwordsDifferent": "New password must be different than the current password",
  },
  "validationError": "Validation Error",
}
`;

exports[`buildNestedI18n should handle Español language 1`] = `
{
  "changePassword": {
    "changePasswordButton": "Cambiar contraseña",
  },
  "common": {
    "changePassword": "Cambiar contraseña",
    "confirmNewPassword": "Confirmar nueva contraseña",
    "currentPassword": "Contraseña actual",
    "dashboard": "Tablero",
    "loading": "Cargando...",
    "logo": "logo",
    "newPassword": "Nueva contraseña",
    "returnToKitchen": "Volver a la Cocina",
    "site": "Chili y Cilantro",
    "startCooking": "Comenzar a cocinar",
    "tagline": "Un Juego de Bluff Picante",
    "unauthorized": "No autorizado",
    "unexpectedError": "Error inesperado",
  },
  "dashboard": {
    "gamesCreated": "Juegos que has creado",
    "gamesParticipating": "Juegos en los que participas",
    "noGames": "No hay juegos disponibles.",
    "title": "Tu Tablero",
  },
  "forgotPassword": {
    "title": "Contraseña olvidada",
  },
  "game": {
    "createGame": "Crear juego",
    "listSuccess": "Lista de juegos recuperada con éxito",
  },
  "keyFeatures": {
    "1": "Emocionante juego de faroleo con un toque culinario",
    "2": "Pujas estratégicas y colocación de cartas",
    "3": "Rápido de aprender, desafiante de dominar",
    "4": "Admite 2 o más jugadores",
    "5": "Rondas de volteo de cartas llenas de suspenso",
    "6": "Gestión de riesgos: ¡evita el temido chile!",
    "7": "El primero en condimentar dos platos o el último chef en pie gana",
    "8": "Perfecto para noches de juegos y reuniones familiares",
    "title": "Características Clave",
  },
  "languageUpdate": {
    "success": "Idioma actualizado con éxito",
  },
  "login": {
    "loginButton": "Iniciar sesión",
  },
  "logoutButton": "Cerrar sesión",
  "registerButton": "Registrarse",
  "splash": {
    "description": "En Chili and Cilantro, los aspirantes a chef compiten para crear el plato perfecto. Tu objetivo es agregar la cantidad justa de cilantro sin arruinarlo con un chile abrasador. ¡Sé el primero en condimentar con éxito dos platos o sé el último chef en pie para ganar!",
    "howToPlay": "Como Jugar",
  },
  "validation": {
    "confirmNewPassword": "Se requiere la confirmación de la nueva contraseña",
    "currentPasswordRequired": "Se requiere la contraseña actual",
    "invalidEmail": "Correo electrónica inválido",
    "invalidLanguage": "Idioma inválido",
    "invalidToken": "Token inválido",
    "newPasswordRequired": "Se requiere la nueva contraseña",
    "passwordMatch": "La contraseña y la confirmación deben coincidir",
    "passwordRegexError": "La contraseña debe tener al menos 8 caracteres y incluir al menos una letra, un número y un carácter especial (!@#$%^&*()_+-=[]{};':"|,.<>/?)",
    "passwordsDifferent": "La nueva contraseña debe ser diferente a la contraseña actual",
  },
  "validationError": "Error de validación",
}
`;

exports[`buildNestedI18n should handle Français language 1`] = `
{
  "changePassword": {
    "changePasswordButton": "Changer le mot de passe",
  },
  "common": {
    "changePassword": "Changer le mot de passe",
    "confirmNewPassword": "Confirmer le nouveau mot de passe",
    "currentPassword": "Mot de passe actuel",
    "dashboard": "Tableau de bord",
    "loading": "Chargement...",
    "logo": "logo",
    "newPassword": "Nouveau mot de passe",
    "returnToKitchen": "Retour à la Cuisine",
    "site": "Piment et Coriandre",
    "startCooking": "Commencer la Cuisine",
    "tagline": "Un Jeu de Bluff Épicé",
    "unauthorized": "Non autorisé",
    "unexpectedError": "Erreur inattendue",
  },
  "dashboard": {
    "gamesCreated": "Jeux que vous avez créé",
    "gamesParticipating": "Jeux que vous participez",
    "noGames": "Aucun jeu disponible.",
    "title": "Votre Tableau de Bord",
  },
  "forgotPassword": {
    "title": "Mot de passe oublié",
  },
  "game": {
    "createGame": "Créer un Jeu",
    "listSuccess": "Liste de jeux récupérée avec succès",
  },
  "keyFeatures": {
    "1": "Un jeu de bluff passionnant avec une touche culinaire",
    "2": "Enchères stratégiques et placement de cartes",
    "3": "Rapide à apprendre, difficile à maîtriser",
    "4": "Prend en charge 2 joueurs ou plus",
    "5": "Rondes de retournement de cartes pleines de suspense",
    "6": "Gestion des risques : évitez le redoutable chili !",
    "7": "Le premier à assaisonner deux plats ou le dernier chef debout gagne",
    "8": "Parfait pour les soirées de jeux et les réunions de famille",
    "title": "Caractéristiques Principales",
  },
  "languageUpdate": {
    "success": "Langue mise à jour avec succès",
  },
  "login": {
    "loginButton": "Connexion",
  },
  "logoutButton": "Déconnexion",
  "registerButton": "S'inscrire",
  "splash": {
    "description": "Dans Chili and Cilantro, les chefs en herbe rivalisent pour créer le plat parfait. Votre objectif est d'ajouter juste la bonne quantité de coriandre sans gâcher le tout avec un piment brûlant. Soyez le premier à réussir à assaisonner deux plats ou soyez le dernier chef debout pour gagner !",
    "howToPlay": "Comment Jouer",
  },
  "validation": {
    "confirmNewPassword": "La confirmation du nouveau mot de passe est requise",
    "currentPasswordRequired": "Le mot de passe actuel est requis",
    "invalidEmail": "Courriel invalide",
    "invalidLanguage": "Langue invalide",
    "invalidToken": "Jeton invalide",
    "newPasswordRequired": "Le nouveau mot de passe est requis",
    "passwordMatch": "Le mot de passe et la confirmation doivent correspondre",
    "passwordRegexError": "Le mot de passe doit comporter au moins 8 caractères et inclure au moins une lettre, un chiffre et un caractère spécial (!@#$%^&*()_+-=[]{};':"|,.<>/?)",
    "passwordsDifferent": "Le nouveau mot de passe doit être différent du mot de passe actuel",
  },
  "validationError": "Erreur de validation",
}
`;

exports[`buildNestedI18n should handle Український language 1`] = `
{
  "changePassword": {
    "changePasswordButton": "Змінити пароль",
  },
  "common": {
    "changePassword": "Змінити пароль",
    "confirmNewPassword": "Підтвердити новий пароль",
    "currentPassword": "Поточний пароль",
    "dashboard": "Панель",
    "loading": "Завантаження...",
    "logo": "лого",
    "newPassword": "Новий пароль",
    "returnToKitchen": "Повернутися на Кухню",
    "site": "Чілі та Коріандр",
    "startCooking": "Почати приготування",
    "tagline": "Пікантний Блеф",
    "unauthorized": "Немає авторизації",
    "unexpectedError": "Неочікувана помилка",
  },
  "dashboard": {
    "gamesCreated": "Гри, які ти створив",
    "gamesParticipating": "Гри, в яких ти береш участь",
    "noGames": "Немає доступних ігор.",
    "title": "Ваша Панель",
  },
  "forgotPassword": {
    "title": "Забули пароль",
  },
  "game": {
    "createGame": "Створити гру",
    "listSuccess": "Список ігор успішно отримано",
  },
  "keyFeatures": {
    "1": "Захоплюючий геймплей блефу з кулінарним родзинкою",
    "2": "Стратегічні ставки та розміщення карток",
    "3": "Швидко навчатися, складно освоїти",
    "4": "Підтримує 2 або більше гравців",
    "5": "Раунди напруженого гортання карт",
    "6": "Управління ризиками: уникайте страшного чилі!",
    "7": "Перемагає перший, хто приправив дві страви, або останній шеф-кухар",
    "8": "Ідеально підходить для вечірніх ігор і сімейних зустрічей",
    "title": "Ключові Жарактеристики",
  },
  "languageUpdate": {
    "success": "Мова оновлена успішно",
  },
  "login": {
    "loginButton": "Увійти",
  },
  "logoutButton": "Вийти",
  "registerButton": "Зареєструватися",
  "splash": {
    "description": "У Chili and Cilantro починаючі кухарі змагаються, щоб створити ідеальну страву. Ваша мета — додати потрібну кількість кінзи, не зіпсувавши її пекучим чилі. Будьте першим, хто успішно приправить дві страви, або будьте останнім шеф-кухарем, який виграє!",
    "howToPlay": "Як Грати",
  },
  "validation": {
    "confirmNewPassword": "Підтвердження нового пароля є обов’язковим",
    "currentPasswordRequired": "Поточний пароль є обов’язковим",
    "invalidEmail": "Недійсний електронній адрес",
    "invalidLanguage": "Недійсна мова",
    "invalidToken": "Недійсний токен",
    "newPasswordRequired": "Новий пароль є обов’язковим",
    "passwordMatch": "Пароль та підтвердження повинні співпадати",
    "passwordRegexError": "Пароль повинен містити принаймні 8 символів, включаючи принаймні одну літеру, одну цифру та один спеціальний символ (!@#$%^&*()_+-=[]{};':"|,.<>/?)",
    "passwordsDifferent": "Новий пароль повинен відрізнятися від поточного",
  },
  "validationError": "Помилка валідації",
}
`;

exports[`buildNestedI18n should handle 中文 language 1`] = `
{
  "changePassword": {
    "changePasswordButton": "更改密码",
  },
  "common": {
    "changePassword": "更改密码",
    "confirmNewPassword": "确认新密码",
    "currentPassword": "当前密码",
    "dashboard": "仪表板",
    "loading": "加载中...",
    "logo": "标志",
    "newPassword": "新密码",
    "returnToKitchen": "返回厨房",
    "site": "辣椒和香菜",
    "startCooking": "开始烹饪",
    "tagline": "一款辛辣的虚张声势游戏",
    "unauthorized": "未经授权",
    "unexpectedError": "意外错误",
  },
  "dashboard": {
    "gamesCreated": "您创建的游戏",
    "gamesParticipating": "您参与的游戏",
    "noGames": "没有可用的游戏。",
    "title": "您的仪表板",
  },
  "forgotPassword": {
    "title": "忘记密码",
  },
  "game": {
    "createGame": "创建游戏",
    "listSuccess": "成功检索游戏列表",
  },
  "keyFeatures": {
    "1": "令人兴奋的虚张声势游戏，带有烹饪的扭曲",
    "2": "战略性出价和卡片放置",
    "3": "易学，难以掌握",
    "4": "支持2个或更多玩家",
    "5": "一轮轮充满悬念的纸牌游戏",
    "6": "风险管理：避免可怕的辣椒！",
    "7": "首先调味两道菜或最后一位厨师站起来赢得胜利",
    "8": "适合游戏之夜和家庭聚会",
    "title": "主要特点",
  },
  "languageUpdate": {
    "success": "语言更新成功",
  },
  "login": {
    "loginButton": "登录",
  },
  "logoutButton": "注销",
  "registerButton": "注册",
  "splash": {
    "description": "在《辣椒和香菜》中，有抱负的厨师们竞相制作完美的菜肴。你的目标是添加适量的香菜，而不会让辣椒烧焦。成为第一个成功调味两道菜的人，或者成为最后一个获胜的厨师",
    "howToPlay": "如何玩",
  },
  "validation": {
    "confirmNewPassword": "确认新密码是必需的",
    "currentPasswordRequired": "当前密码是必需的",
    "invalidEmail": "无效电子邮件",
    "invalidLanguage": "无效语言",
    "invalidToken": "无效令牌",
    "newPasswordRequired": "新密码是必需的",
    "passwordMatch": "密码和确认必须匹配",
    "passwordRegexError": "密码必须至少8个字符长且包含至少一个字母、一个数字和一个特殊字符(!@#$%^&*()_+-=[]{};:"|,.<>/?)",
    "passwordsDifferent": "新密码必须不同于当前密码",
  },
  "validationError": "验证错误",
}
`;
